src/main/java/org/spongepowered/api/command/parameter/managed/ValueUsage.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.parameter.managed;
+
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.text.Text;
+
+/**
+ * Defines the usage string for the parameter.
+ */
+@FunctionalInterface
+public interface ValueUsage {
+
+    /**
+     * Gets the usage string for the argument.
+     *
+     * @param cause The {@link Cause} requesting the usage
+     * @param key The {@link Text} that defines the parameter key
+     * @return The usage
+     */
+    Text getUsage(Cause cause, Text key);
```

Comment:
Maybe `Parameter.Key parameterKey` instead of `Text key`?

Reply:
No, as it's passing in how the command processor will render it, which would be more useful to the consumer as the API doesn't define what the usage would look like, the implementation does. This method can then inspect what the current state is and make a decision based on that, rather than just the key.